collect/queue.go

```diff
@@ -136,3 +111,26 @@ func recordSent(num int) {
 	sent += int64(num)
 	slock.Unlock()
 }
+
+func SendDataPoints(dps []*opentsdb.DataPoint, tsdb string) (*http.Response, error) {
+	var buf bytes.Buffer
+	g := gzip.NewWriter(&buf)
+	if err := json.NewEncoder(g).Encode(dps); err != nil {
+		return nil, err
+	}
+	if err := g.Close(); err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("POST", tsdb, &buf)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Content-Encoding", "gzip")
+
+	resp, err := client.Do(req)
+	if err == nil {
+		defer resp.Body.Close()
```

Comment:
This breaks the error handling in `send()`, which reads from the body in case the status code isn't 204.